### ISSUE TITLE
Allow dynamic route for index pages with optional parameter

### ISSUE THREAD

**Issue by mklueh** [nuxt/nuxt#15028]
### Describe the feature

I don't know if this is a feature or rather a bug.

My blog has pagination on it's root page and I need to make the dynamic page parameter optional.
However, when I do not provide any route parameter, it will lead to a 404 error immediately.

In my opinion this should be possible, because otherwise there might be no other option than making a redirect from the root to the root + route param, which is bad for SEO as far as I know.

There is already a discussion, but I've decided to better create an issue as well

https://github.com/nuxt/framework/discussions/7526

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by GraniteEducation** [comment]
@danielroe I am experiencing a similar issue. I have the page/route structure below:

-[course_id]
   --index.vue
   --[[mode]]
   
   With this structure, I am never able to access my index.vue page. 
   visiting: 
/somecourseid (route #1)
/somecourseid/build_mode (route #2)

bring me always brings my users into the pages defined under the "mode" optional parameter. With the "mode" parameter set to (I believe) an empty string (route #1). I have tried manually setting the "mode" parameter to null and that did not fix the issue either. 

In short, it seems that the Nuxt  is **NOT evaluating optional params last** - as is described in the documentation. 

Thank you in advance for any help!

**Comment by danielroe** [comment]
@GraniteEducation Could you provide a minimal reproduction? 🙏

**Comment by GraniteEducation** [comment]
Sure thing! Here is a link. I used Stackblitz so you can run it: 

https://stackblitz.com/edit/github-gehy7r?file=pages%2Findex.vue

**Comment by danielroe** [comment]
@GraniteEducation shouldn't you be using this structure?

```
-[course_id]
--index.vue
--[mode]
```

(Because `[mode]` is not optional - if it's missing you want `index.vue` to be displayed instead.)

**Comment by GraniteEducation** [comment]
@danielroe thank you so much for looking into this. I was running into a scenario where I have a route with the "mode" set to build, and I needed to "drop" that parameter to return to index. When I had [mode] set to required and was trying to push a route with a "null" [mode], e.g. router.push({params: {mode: null}} it was giving me an error stating that that "mode was a required parameter". I assumed that making the "parameter optional" would allow me the ability to nullify it. 

I see that my interpretation of param nullification is made possible through param optionality was likely wrong. My apologies for any time my confusion costed you. 

Have a wonderful day!
